app/app.go

```diff
@@ -140,3 +149,63 @@ func addData(management *config.ManagementContext, cfg Config) error {
 
 	return addMachineDrivers(management)
 }
+
+func getZombiePid(path string) int {
+	// Ignore errors
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return 0
+	}
+
+	fields := strings.Split(string(content), ") ")
+	fields = strings.Split(fields[len(fields)-1], " ")
+
+	if fields[0] == "Z" && fields[1] == "1" {
+		i, _ := strconv.Atoi(strings.Split(string(content), " ")[0])
+		return i
+	}
+
+	return 0
+}
+
+func reaper() {
+	reap := make([]string, 0, 5)
+
+	for {
+		time.Sleep(1 * time.Second)
+
+		for _, zombie := range reap {
+			zombiePid := getZombiePid(zombie)
+			logrus.Debugf("Reaping PID %s : %d", zombie, zombiePid)
+			if zombiePid <= 0 {
+				continue
+			}
+
+			reaped, err := syscall.Wait4(zombiePid, nil, syscall.WNOHANG, nil)
+			if err != nil || reaped <= 0 {
+				logrus.Errorf("Failed to reap %d, got %v: %v", zombiePid, reaped, err)
+			}
+		}
+
+		reap = reap[:0]
+
+		files, err := filepath.Glob("/proc/*/stat")
+		if err != nil {
+			logrus.Errorf("Failed to read processes : %v", err)
+			continue
+		}
+
+		for _, file := range files {
+			if getZombiePid(file) > 0 {
+				reap = append(reap, file)
+			}
+		}
+	}
+}
+
+// NewDaemon sets up everything for the daemon to be able to service
+// requests from the webserver.
+func NewDaemon() {
```

Comment:
NewDaemon is not a good name. Can you rename it something more descriptive?
Also, only export it if it is being called outside of this package